10524 upgrade bootstrap v5

roseeichelmann opened the pull request:
Fixes cityofaustin/atd-data-tech/issues/10524

This PR upgrades bootstrap to v5 and react-bootstrap to the corresponding v2.5. There were a lot of breaking changes with these upgrades but I think I combed through and fixed them all, let me know if there is something I missed! I got this iteration of the website looking super close to the original aside from the padding being a tiny bit off, I can't seem to get everything to align exactly as it was before but I think this is okay? We are also about to revamp the website so I figured I shouldn't waste too much time being a perfectionist if we will be changing a lot.

https://getbootstrap.com/docs/5.0/migration/
https://react-bootstrap.github.io/migrating/

Steps to test:

[Netlify preview](https://deploy-preview-49--atd-product.netlify.app/)

1. You can compare the netlify deploy to the current website to check if there are any big differences between them that need to be corrected.
2. Check the Home page, Our work, What we deliver, About us pages, along with the projects details, products details, click on all the tabs and filters, make sure everything is working/looks good!

roseeichelmann commented on components/projectsView/ProjectFilters.js:
```diff
@@ -76,16 +74,20 @@ export default function ProjectFilters({
+          <Row
+            className={`h-100`}
+            style={{
+              backgroundColor: "#f8f9fa",
```
this `bg-light` doesn't work anymore with the upgrade so I had to add the light grey background with styling

roseeichelmann commented on components/projectsView/ProjectFilters.js:
```diff
@@ -76,16 +74,20 @@ export default function ProjectFilters({
+          >
             <Col>
-              <Form.Control
+              <Form.Select
```
Bootstrap got rid of the down arrow with Form.Control and prefer to use Form.Select, so the `as="select"` prop below can be removed as well

roseeichelmann commented on components/projectsView/ProjectsList.js:
```diff
@@ -121,7 +121,7 @@ export default function ProjectsList(props) {
       {!showChartView && (
         <Row>
-          <Col className="text-right text-muted">
+          <Col className="text-end text-muted">
```
every instance of `right` and `left` needs to be changed to `end` and `start`, respectively

roseeichelmann commented on components/shared/Footer.js:
```diff
@@ -50,20 +50,20 @@ export default function Footer({ links }) {
               />
               <div
-                className="d-flex flex-column align-items-start text-white ml-3"
+                className="d-flex flex-column align-items-start text-white ms-3"
```
also l/r is now s/e (start and end)

roseeichelmann commented on components/shared/Footer.js:
```diff
@@ -50,20 +50,20 @@ export default function Footer({ links }) {
                   Data & Technology Services
                 </span>
-                <span className="font-weight-light py-1">
+                <span className="fs-6 fw-light py-1">
```
upgrade changed font-weight to fw

roseeichelmann commented on components/shared/indexIssueDetails/Comments.js:
```diff
@@ -21,7 +21,7 @@ function Comment({ comment }) {
       <Card.Header>
         <b>{comment.user.login}</b>
-        <span className="text-muted">commented on {createdAt}</span>
+        <span className="text-muted ms-1">commented on {createdAt}</span>
```
add spacing between the username and "commented on.." in activity tab

mddilley replied:
Nice detail!

roseeichelmann commented on components/shared/indexIssueDetails/IndexIssueDetails.js:
```diff
@@ -133,6 +133,7 @@ function InfoRow({ issue }) {
         <h6 className="mb-0 mt-2 text-muted">Issue</h6>
         <a
           href={`https://github.com/cityofaustin/atd-data-tech/issues/${issue.number}`}
+          className="text-primary text-decoration-none"
```
this makes the link our primary dark blue color and removes and underlining/text decoration

roseeichelmann commented on components/shared/indexIssueDetails/IndexIssueDetails.js:
```diff
@@ -167,7 +168,7 @@ export default function IndexIssueDetails({
       <BackLink indexType={indexType} history={router} />
       <h4 className="text-secondary mb-0">
-        {indexType === "project" ? "Project details" : "Product Details"}
+        {indexType === "project" ? "Project Details" : "Product Details"}
```
It bothered me that these weren't both capitalized

johnclary replied:
nice catch! there is some tension around here wrt to title casing so i may have been splitting the difference 🤪  @patrickm02L @amenity

patrickm02L replied:
Capitalize both. There are other examples on the page that makes that capitalization legit.

**_examples_**

<img width="570" alt="Screen Shot 2022-10-28 at 7 41 59 AM" src="https://user-images.githubusercontent.com/5566727/198589407-73e893e8-4e21-4b6f-a3c6-9407b835d197.png">

<img width="462" alt="Screen Shot 2022-10-28 at 7 42 03 AM" src="https://user-images.githubusercontent.com/5566727/198589437-7fc3218d-184f-4744-95f8-9b1e2c9c2c75.png">

roseeichelmann commented on components/projectsView/ProjectFilters.js:
```diff
@@ -76,16 +74,20 @@ export default function ProjectFilters({
+          <Row
+            className={`h-100`}
+            style={{
+              backgroundColor: "#f8f9fa",
```
using `bg-light` as a className to change the background color isn't working anymore with the upgrade so I had to change the background color with styling

johnclary replied:
i think what's going on here is that the Bootstrap scss customizations changed a bit with v5. we are overriding `$theme-colors` is our `custom.scss` file. for example, `bg-primary` works, because it's defined there. so adding an entry for `light` should fix this!

roseeichelmann commented on components/shared/IndexIssueListItem.js:
```diff
@@ -23,7 +23,6 @@ export default function IndexIsssueListItem({ type, issue }) {
   const [description, img] = parseBody(issue.body);
   return (
     <Link
-      className="text-primary text-decoration-none"
```
i changed the default styling of links in the css file to make them all default to this so we don't have to write it everywhere

mddilley replied:
Your update makes link styling more consistent which I fully support. I think this is a great improvement. 👏  

Previously, links in some places underlined on hover while others did not - like the links in the paragraph below the "Our Work " header on `/projects?status=in_progress`. Now, these behave like the others.

roseeichelmann commented on styles/custom.scss:
```diff
@@ -3,6 +3,30 @@ $theme-colors: (
+    sans-serif,
+    // Emoji fonts
+    "Apple Color Emoji",
+    "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji" !important;
```
this part i'm not sure about - the upgrade changed the font of body across the website so i added this to make the `font-family` the same as it was before. is this okay?

mddilley replied:
I looked at this locally with and without this override so that I could try to answer this, and I'm seeing the same font either way. Could be an OS thing?

roseeichelmann commented on styles/custom.scss:
```diff
@@ -70,19 +94,7 @@ $theme-colors: (
-}
-
-.ms-1 {
-  margin-left: 0.25rem;
```
this was from my last PR but is not needed anymore with the upgrades

roseeichelmann commented on styles/custom.scss:
```diff
@@ -101,6 +113,16 @@ $theme-colors: (
+  .container-md,
+  .container-sm,
+  .container-xl {
+    max-width: 1140px !important;
```
the upgrades changed the default width of the containers so that the content was stretching across the entire page instead of having a good amount of white space on the left and right like the website currently has, so i added this to revert back to how the website looked before.

roseeichelmann commented on styles/custom.scss:
```diff
@@ -3,6 +3,30 @@ $theme-colors: (
+$link-color: #0d3e79;
+$hr-color: #dee2e6;
+$hr-opacity: 1;
+$link-decoration: none;
```
here i am overriding some of the default variables to get back to our previous design + our links are styled the same throughout the website so this will take less code and keep us consistent

frankhereford reviewed (approved):
![10](https://user-images.githubusercontent.com/3653206/197843707-eb6a8b9a-90a5-4a48-b675-389dcbb804bc.gif)

Wouldn't change a thing - this looks great. Thanks for all the work that goes into a refactor like this!

mddilley reviewed (approved):
Awesome! I'm behind the curve on this version of Bootstrap so this was a good one to review for me. Some surprising changes in the framework IMO as an old school Bootstrap user. 😄 I added a few comments, but I think the small differences that you noted are just fine. 👍

roseeichelmann commented:
@mddilley yeah, i think it may be a weird linux thing that the font is different for me? i'm gonna go ahead and remove it!

johnclary reviewed (approved):
amazing work, @roseeichelmann! i really appreciate your attention to detail. this was a big undertaking and you made it look like a breeze. i'm so glad to see these libraries brought up to date (and to match the Data. & Performance Hub).

i'd prefer to see the `$theme-colors` updated so we can nix those inline background color styles, but i defer to y'all and am approving.

🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢

roseeichelmann commented:
> amazing work, @roseeichelmann! i really appreciate your attention to detail. this was a big undertaking and you made it look like a breeze. i'm so glad to see these libraries brought up to date (and to match the Data. & Performance Hub).
> 
> i'd prefer to see the `$theme-colors` updated so we can nix those inline background color styles, but i defer to y'all and am approving.
> 
> ship ship ship ship ship ship ship ship ship ship

adding that `light` variable to `$theme-colors` worked like a charm so i removed the inline background styling in favor of bg-light and i also removed the undefined classnames you pointed out. thanks so much for the helpful comments!

patrickm02L reviewed (approved):
Roll it out! 🤘
